BindingContext/BPoP.md

```diff
@@ -0,0 +1,313 @@
+    - [JS API based model:](#js-api-based-model)
+  - [Browser BPoP proofs](#browser-bpop-proofs)
+- [Detailed design](#detailed-design)
+  - [Storage model](#storage-model)
+  - [Retrieval model](#retrieval-model)
+  - [Application model](#application-model)
+    - [BPoP key verification](#bpop-key-verification)
+    - [BPoP background refresh](#bpop-background-refresh)
+  - [Server challenge](#server-challenge)
+  - [Server update](#server-update)
+- [Considered alternatives](#considered-alternatives)
+  - [TLS Token Binding](#tls-token-binding)
+- [Stakeholder Feedback / Opposition](#stakeholder-feedback--opposition)
+- [References & acknowledgements](#references--acknowledgements)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Introduction
+
+The primary use case for BPoP is binding an authentication cookie. Cookies remain among the most common mechanism web servers use to store authentication state about a user. Malicious actors steal such authentication cookies and compromise user data. 
+
+The motivation for BPoP closely follows the motivations for [IETF DPoP](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop), i.e. "to prevent unauthorized or illegitimate parties from using leaked or stolen access tokens, by binding a token to a public key upon issuance and requiring that the client proves possession of the corresponding private key when using the token [...]" except that, rather than binding an access token (issued by an identity provider), BPoP binds a browser artifact (such as a cookie) issued by a website.
+
+This document makes direct analogs to DPoP, e.g. defining a "BPoP Proof" to match DPoP's "DPoP Proof".
+
+## Goals [or Motivating Use Cases, or Scenarios]
+
+- BPoP introduces a mechanism to bind authentication cookies to the user device and hence render the stolen artifacts useless.
+- BPoP aims at securing sub-domains and enables the organization to control the binding boundaries among the sub-domains supported, 
+
+## Non-goals
+
+- BPoP cannot prevent temporary access to the app generated cookies before they are bound. 
+- BPoP also tries to minimize the round trips needed to secure cookies, however can introduce latency/extra time to function.
+
+## BPoP functionality
+
+### Usecases
+
+A website that is its own standalone identity provider (i.e. a website that accepts a username and password) could activate BPoP as part of rending the login form. Then, on the subsequent request, when the website verifies the username and password and issues an authentication cookie, the website could also verify the BPoP proof and record the public key associated with the BPoP proof in the authentication cookie. If this website had user submitted content and such content was subsequently used as part of a stored cross site scripting (XSS) attack, this attack would be unable to steal the BPoP private key and thus the attacker would be unable to use any stolen cookies.
```

Comment:
Perhaps call out man-in-the-middle (someone intercepts the traffic) and on device attacks (someone steals entire browser cookie jar) in addition to XSS as well?

Reply:
Done